### PR TITLE
[TEST] Use in-memory catalog in most test suites

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

Followup #707, use in-memory catalog in most test suites to avoid derby lock to speed up test suites.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/operation/SparkOperationSuite.scala [hunk 1]
```diff
@@ -33,10 +33,10 @@ import org.apache.spark.sql.types._
 import org.apache.kyuubi.Utils
 import org.apache.kyuubi.engine.spark.WithSparkSQLEngine
 import org.apache.kyuubi.engine.spark.shim.SparkCatalogShim
-import org.apache.kyuubi.operation.JDBCTests
+import org.apache.kyuubi.operation.HiveJDBCTests
 import org.apache.kyuubi.operation.meta.ResultSetSchemaConstant._
 
-class SparkOperationSuite extends WithSparkSQLEngine with JDBCTests {
+class SparkOperationSuite extends WithSparkSQLEngine with HiveJDBCTests {
```
Comment (turboFei) [inline externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/operation/SparkOperationSuite.scala:39]: Do you set catalog implementation to hive in this test suite?

Comment (turboFei) [reply externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/operation/SparkOperationSuite.scala:39]: For HiveJDBCTests, it is only test for hive catalog?

Comment (pan3793) [reply externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/operation/SparkOperationSuite.scala:39]: hive is the default catalogImplementation
```
    val defaultCat = if (KyuubiSparkUtil.hiveClassesArePresent) "hive" else "in-memory"
    sparkConf.setIfMissing("spark.sql.catalogImplementation", defaultCat)
```

Comment (pan3793) [reply externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/operation/SparkOperationSuite.scala:39]: Actually, hive catalog has more functionalities than in-memory catalog, I'm not sure if we need in-memory catalog test in both engine side and kyuubi server side.

File: kyuubi-common/src/test/scala/org/apache/kyuubi/operation/HiveJDBCTests.scala [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.operation
+
+import org.apache.kyuubi.Utils
+import org.apache.kyuubi.operation.meta.ResultSetSchemaConstant._
+
+trait HiveJDBCTests extends JDBCTestUtils {
```
Comment (yaooqinn) [inline kyuubi-common/src/test/scala/org/apache/kyuubi/operation/HiveJDBCTests.scala:23]: this base trait mainly duplicated, can we make it neater?

Comment (pan3793) [reply kyuubi-common/src/test/scala/org/apache/kyuubi/operation/HiveJDBCTests.scala]: updated, let trait HiveJDBCTests extends BasicJDBCTests

File: kyuubi-common/src/test/scala/org/apache/kyuubi/operation/HiveJDBCTests.scala [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.operation
+
+import org.apache.kyuubi.Utils
+import org.apache.kyuubi.operation.meta.ResultSetSchemaConstant._
+
+trait HiveJDBCTests extends BasicJDBCTests {
```
Comment (yaooqinn) [inline kyuubi-common/src/test/scala/org/apache/kyuubi/operation/HiveJDBCTests.scala:23]: HiveCatalogTests

**Review by yaooqinn — APPROVED** [review]
